stan-dev/stan
# Feature/issue 1244 parser expects

#### Summary:

Clean up error messages in the parser to make them more understandable for users, remove duplicate output, and to provide errors and reasons in the places where they were missing.

#### Intended Effect:

Make it easier for users to debug Stan programs syntactically.

#### How to Verify:

There are a bunch of new test models that make sure the parser produces the right error.  To get the long form, use stanc to parse the models.  

#### Side Effects:

Error messages change text, which may affect unit tests for CmdStan or other interfaces.

#### Documentation:

Nothing user facing.

#### Reviewer Suggestions:

Anyone.
